src/6_branch.js

```diff
@@ -541,6 +541,39 @@ Branch.prototype['init'] = wrap(
 );
 
 /**
+ * @function Branch.deepview
+ * @param {function(?Error, utils.sessionData=)=} callback - _optional_ - callback to read the session data.
+ */
+Branch.prototype['deepview'] = wrap(callback_params.CALLBACK_ERR_DATA, function(done, data, options) {
+	var self = this;
+
+	if (!options) {
+		options = { };
+	}
+	var cleanedData = utils.cleanLinkData(data);
+
+	if (options['open_app']) {
+		cleanedData['open_app'] = true;
+	}
+
+	var referringLink = self._referringLink();
+	if (referringLink && !options['make_new_link']) {
+		cleanedData['link_click_id'] = referringLink.substring(
+			referringLink.lastIndexOf('/') + 1, referringLink.length
+		);
+	}
+
+	this._api(resources.deepview, cleanedData, function(err, data) {
+		if (err) {
+			done(err, data);
+		}
+
+		self._server.createScript(data[0], true /* internal */);
```

Comment:
ah - we can do this in an even simpler way: just treat the API request as a jsonp request, that way it gets put in a JS embed. then we just have to be sure to actually call the callback in the js that's returned from the server. does that make sense?